File: frontend/src/api/aitestrebort/project.ts
/**
 * aitestrebort 项目管理 API
 * 基于 aitestrebort 项目管理功能
 */
import { baseDiraitestrebort } from '@/api/base-url'
import request from '@/utils/system/request'

export interface Project {
  id: number
  name: string
  description: string
  creator_id: number
  created_at: string
  updated_at: string
}

export interface ProjectCredential {
  id: number
  system_url: string
  username: string
  password?: string
  user_role: string
  created_at: string
}

export interface ProjectMember {
  id: number
  user_id: number
  role: 'admin' | 'developer' | 'tester' | 'viewer'
  created_at: string
}

export interface CreateProjectData {
  name: string
  description?: string
}

export interface CreateCredentialData {
  system_url?: string
  username?: string
  password?: string
  user_role?: string
}

export interface CreateMemberData {
  user_id: number
  role: 'admin' | 'developer' | 'tester' | 'viewer'
}

export interface LLMConfig {
  id: number
  name: string
  provider: string
  model_name: string
  api_key: string
  base_url?: string
  temperature?: number
  max_tokens?: number
  is_default: boolean
  created_at: string
}

export interface CreateLLMConfigData {
  name: string
  provider: string
  model_name: string
  api_key: string
  base_url?: string
  temperature?: number
  max_tokens?: number
  is_default?: boolean
}

export interface MCPConfig {
  id: number
  name: string
  url: string
  transport: string
  headers?: Record<string, string>
  is_enabled: boolean
  created_at: string
}

export interface CreateMCPConfigData {
  name: string
  url: string
  transport?: string
  headers?: Record<string, string>
  is_enabled?: boolean
}

export interface APIKey {
  id: number
  name: string
  service_type: string
  api_key: string
  description?: string
  is_active: boolean
  created_at: string
}

export interface CreateAPIKeyData {
  name: string
  service_type: string
  api_key: string
  description?: string
  is_active?: boolean
}

export interface Conversation {
  id: number
  title: string
  llm_config_id?: number
  created_at: string
  updated_at: string
}

export interface ConversationMessage {
  id: number
  conversation_id: number
  role: 'user' | 'assistant' | 'system'
  content: string
  created_at: string
}

export interface CreateConversationData {
  title: string
  llm_config_id?: number
}

export interface SendMessageData {
  content: string
  role?: 'user' | 'system'
}

// 项目管理 API
export const projectApi = {
  // 获取项目列表
  getProjects: (params?: {
    search?: string
    page?: number
    page_size?: number
  }) => {
    return request.get(`${baseDiraitestrebort}/projects`, { params })
  },

  // 获取项目详情
  getProject: (id: number) => {
    return request.get(`${baseDiraitestrebort}/projects/${id}`)
  },

  // 创建项目
  createProject: (data: CreateProjectData) => {
    return request.post(`${baseDiraitestrebort}/projects`, data)
  },

  // 更新项目
  updateProject: (id: number, data: Partial<CreateProjectData>) => {
    return request.put(`${baseDiraitestrebort}/projects/${id}`, data)
  },

  // 删除项目
  deleteProject: (id: number) => {
    return request.delete(`${baseDiraitestrebort}/projects/${id}`)
  },

  // 项目凭据管理
  getCredentials: (projectId: number) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/credentials`)
  },

  createCredential: (projectId: number, data: CreateCredentialData) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/credentials`, data)
  },

  updateCredential: (projectId: number, credentialId: number, data: Partial<CreateCredentialData>) => {
    return request.put(`${baseDiraitestrebort}/projects/${projectId}/credentials/${credentialId}`, data)
  },

  deleteCredential: (projectId: number, credentialId: number) => {
    return request.delete(`${baseDiraitestrebort}/projects/${projectId}/credentials/${credentialId}`)
  },

  // 项目成员管理
  getMembers: (projectId: number) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/members`)
  },

  addMember: (projectId: number, data: CreateMemberData) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/members`, data)
  },

  updateMember: (projectId: number, memberId: number, data: { role: string }) => {
    return request.put(`${baseDiraitestrebort}/projects/${projectId}/members/${memberId}`, data)
  },

  removeMember: (projectId: number, memberId: number) => {
    return request.delete(`${baseDiraitestrebort}/projects/${projectId}/members/${memberId}`)
  },

  // LLM 配置管理
  getLLMConfigs: (projectId: number, params?: {
    search?: string
    provider?: string
    is_active?: boolean
    page?: number
    page_size?: number
  }) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/llm-configs`, { params })
  },

  createLLMConfig: (projectId: number, data: CreateLLMConfigData) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/llm-configs`, data)
  },

  updateLLMConfig: (projectId: number, configId: number, data: Partial<CreateLLMConfigData>) => {
    return request.put(`${baseDiraitestrebort}/projects/${projectId}/llm-configs/${configId}`, data)
  },

  deleteLLMConfig: (projectId: number, configId: number) => {
    return request.delete(`${baseDiraitestrebort}/projects/${projectId}/llm-configs/${configId}`)
  },

  // MCP 配置管理
  getMCPConfigs: (projectId: number, params?: {
    search?: string
    is_enabled?: boolean
    page?: number
    page_size?: number
  }) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/mcp-configs`, { params })
  },

  createMCPConfig: (projectId: number, data: CreateMCPConfigData) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/mcp-configs`, data)
  },

  updateMCPConfig: (projectId: number, configId: number, data: Partial<CreateMCPConfigData>) => {
    return request.put(`${baseDiraitestrebort}/projects/${projectId}/mcp-configs/${configId}`, data)
  },

  deleteMCPConfig: (projectId: number, configId: number) => {
    return request.delete(`${baseDiraitestrebort}/projects/${projectId}/mcp-configs/${configId}`)
  },

  // 测试 MCP 连接
  testMCPConnection: (projectId: number, configId: number) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/mcp-configs/${configId}/test`)
  },

  // API 密钥管理
  getAPIKeys: (projectId: number, params?: {
    search?: string
    service_type?: string
    is_active?: boolean
    page?: number
    page_size?: number
  }) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/api-keys`, { params })
  },

  createAPIKey: (projectId: number, data: CreateAPIKeyData) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/api-keys`, data)
  },

  updateAPIKey: (projectId: number, keyId: number, data: Partial<CreateAPIKeyData>) => {
    return request.put(`${baseDiraitestrebort}/projects/${projectId}/api-keys/${keyId}`, data)
  },

  deleteAPIKey: (projectId: number, keyId: number) => {
    return request.delete(`${baseDiraitestrebort}/projects/${projectId}/api-keys/${keyId}`)
  },

  // LLM 对话管理
  getConversations: (projectId: number) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/conversations`)
  },

  createConversation: (projectId: number, data: CreateConversationData) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/conversations`, data)
  },

  getConversationMessages: (projectId: number, conversationId: number) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/conversations/${conversationId}/messages`)
  },

  sendMessage: (projectId: number, conversationId: number, data: SendMessageData) => {
    return request.post(`${baseDiraitestrebort}/projects/${projectId}/conversations/${conversationId}/messages`, data)
  },

  deleteConversation: (projectId: number, conversationId: number) => {
    return request.delete(`${baseDiraitestrebort}/projects/${projectId}/conversations/${conversationId}`)
  },

  // 提示词管理
  getPrompts: (projectId: number, params?: {
    search?: string
    prompt_type?: string
    is_active?: boolean
    page?: number
    page_size?: number
  }) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/prompts`, { params })
  },

  // 知识库管理
  getKnowledgeBases: (projectId: number, params?: {
    search?: string
    is_active?: boolean
    page?: number
    page_size?: number
  }) => {
    return request.get(`${baseDiraitestrebort}/projects/${projectId}/knowledge-bases`, { params })
  }
}

File: backend/app/routers/aitestrebort/project.py
"""
aitestrebort 项目管理路由
"""
from ..base_view import APIRouter
from ...services.aitestrebort import project as project_service

router = APIRouter()

"""
aitestrebort 项目管理路由
"""
from ..base_view import APIRouter
from ...services.aitestrebort import project as project_service

router = APIRouter()

# 基础项目管理路由
router.add_get_route("/test", project_service.test_simple, auth=False, summary="测试接口")
router.add_get_route("/projects", project_service.get_projects, summary="获取项目列表")
router.add_post_route("/projects", project_service.create_project, summary="创建项目")
router.add_get_route("/projects/{project_id}", project_service.get_project_detail, summary="获取项目详情")
router.add_put_route("/projects/{project_id}", project_service.update_project, summary="更新项目")
router.add_delete_route("/projects/{project_id}", project_service.delete_project, summary="删除项目")
router.add_get_route("/projects/{project_id}/statistics", project_service.get_project_statistics, summary="获取项目统计信息")

# 项目凭据管理路由
router.add_get_route("/projects/{project_id}/credentials", project_service.get_project_credentials, summary="获取项目凭据列表")
router.add_post_route("/projects/{project_id}/credentials", project_service.create_project_credential, summary="创建项目凭据")

# 项目成员管理路由
router.add_get_route("/projects/{project_id}/members", project_service.get_project_members, summary="获取项目成员列表")
router.add_post_route("/projects/{project_id}/members", project_service.add_project_member, summary="添加项目成员")
router.add_delete_route("/projects/{project_id}/members/{member_id}", project_service.remove_project_member, summary="移除项目成员")

# LLM 配置管理路由
router.add_get_route("/projects/{project_id}/llm-configs", project_service.get_llm_configs, summary="获取LLM配置列表")
router.add_post_route("/projects/{project_id}/llm-configs", project_service.create_llm_config, summary="创建LLM配置")
router.add_put_route("/projects/{project_id}/llm-configs/{config_id}", project_service.update_llm_config, summary="更新LLM配置")
router.add_delete_route("/projects/{project_id}/llm-configs/{config_id}", project_service.delete_llm_config, summary="删除LLM配置")

# MCP 配置管理路由
router.add_get_route("/projects/{project_id}/mcp-configs", project_service.get_mcp_configs, summary="获取MCP配置列表")
router.add_post_route("/projects/{project_id}/mcp-configs", project_service.create_mcp_config, summary="创建MCP配置")
router.add_put_route("/projects/{project_id}/mcp-configs/{config_id}", project_service.update_mcp_config, summary="更新MCP配置")
router.add_delete_route("/projects/{project_id}/mcp-configs/{config_id}", project_service.delete_mcp_config, summary="删除MCP配置")
router.add_post_route("/projects/{project_id}/mcp-configs/{config_id}/test", project_service.test_mcp_config, summary="测试MCP配置连接")

# API 密钥管理路由
router.add_get_route("/projects/{project_id}/api-keys", project_service.get_api_keys, summary="获取API密钥列表")
router.add_post_route("/projects/{project_id}/api-keys", project_service.create_api_key, summary="创建API密钥")
router.add_put_route("/projects/{project_id}/api-keys/{key_id}", project_service.update_api_key, summary="更新API密钥")
router.add_delete_route("/projects/{project_id}/api-keys/{key_id}", project_service.delete_api_key, summary="删除API密钥")
router.add_post_route("/projects/{project_id}/api-keys/{key_id}/test", project_service.test_api_key, summary="测试API密钥")
router.add_post_route("/projects/{project_id}/api-keys/{key_id}/regenerate", project_service.regenerate_api_key, summary="重新生成API密钥")

# LLM 对话管理路由
router.add_get_route("/projects/{project_id}/conversations", project_service.get_conversations, summary="获取对话列表")
router.add_post_route("/projects/{project_id}/conversations", project_service.create_conversation, summary="创建对话")
router.add_get_route("/projects/{project_id}/conversations/{conversation_id}/messages", project_service.get_conversation_messages, summary="获取对话消息")
router.add_post_route("/projects/{project_id}/conversations/{conversation_id}/messages", project_service.send_message, summary="发送消息")
router.add_delete_route("/projects/{project_id}/conversations/{conversation_id}", project_service.delete_conversation, summary="删除对话")

# 提示词管理路由
router.add_get_route("/projects/{project_id}/prompts", project_service.get_prompts, summary="获取提示词列表")

# 知识库管理路由（代理到知识库服务）
router.add_get_route("/projects/{project_id}/knowledge-bases", project_service.get_knowledge_bases, summary="获取知识库列表")